docs/source/cpp/acero_overview.rst

```diff
@@ -0,0 +1,29 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+*****
+Acero
+*****
+
+.. toctree::
+   :maxdepth: 3
+
+   acero/overview
```

Comment:
Putting the overview in its entirety, including how to make an ExecPlan, before discussing Substrait increases the risk that a user will use an ExecPlan by hand instead of using Substrait as a helper layer. Could overview be broken up further, and one part put before Substrait and one after? 

Really, it seems what's necessary is a good example of using Substrait to make one of these trees -- right now, all we offer is how to do it directly, which is really helpful for some developers, but drives them away from Substrait.

Reply:
I see your point here. I totally understand that. I think we definitely need to re-write what we offer carefully since Substrait came into the picture. I am planning to work on that piece separately. And before doing that, I thought it is better clean up the documentation strucutre. A while back I created this [JIRA](https://issues.apache.org/jira/browse/ARROW-17210) to address this point. At that time I assumed that this has to be done at a certain point. But Substrait adoption in Acero wasn't matured enough to write a good documentation. But we have improved since that Jira was created.

Shall we address this issue in that JIRA (I can create it righ after re-organization is merged) and just keep the re-organization piece in this PR? WDYT?

cc @westonpace

Reply:
I think both points are fine.  First, I agree with @ksuarez1423 .  The page that gives an overview of the execution engine should link to our substrait docs.  In addition, the part that says "creating exec plans" should have some kind of note or warning that users can use substrait to create more portable execution plans and should only read this section for advanced usage.

Reply:
Oh, but this can also be done in a follow-up PR if we want this one to be focused solely on file reorg.